types/aws-lambda/trigger/eventbridge.d.ts

```diff
@@ -15,4 +15,5 @@ export interface EventBridgeEvent<TDetailType extends string, TDetail> {
     source: string;
     'detail-type': TDetailType;
     detail: TDetail;
+    'replay-name'?: string;
```

Comment:
Skimming some other types in this package, I suspect this should have `| undefined` to support `exactOptionalPropertyTypes`.

Reply:
My understanding is those were automatically converted by MS when they added the option to avoid breaking everyone's existing code (with the option enabled). Otherwise it would never be usable.

I don't think that applies to new types.

Reply:
Strictly, it should be `| undefined` if it can, in fact, be present but undefined, which is not true for JSON-parsed input values and is for JSON-stringified output values.

Reply:
I think we have been pushing for contributions to ensure that things work (@peterblazejewicz definitely has), but we are in sore need for a little more documentation on this in our README as a gotcha.

Reply:
yes, we promote optional usage (not enforced) to provide better interop in ecosystem (for projects using strict features consuming DT definitions) - where it make sense (user input data/configurations/options/React props)

Reply:
To clarify, are all new optional types proposed in DT definitions expected to be in the form `| undefined` as well as `?`, or only if there's a difference in runtime behaviour between `undefined` and 'not present whatsoever' in a given package?

Reply:
If there's a difference in behavior between explicitly undefined and "not present", then you wouldn't want to leave the `?`. If there's no difference, then having both makes things more compatible for those who do enable the flag.

It's not that this is a strict requirement for DT packages; there have certainly been PRs merged that don't make the distinction, and it's not that hard to fix if someone reports that they can't use the types.

I try my best to mention it when I see that the package already has `| undefined`, solely for consistency, but overall we need better reference for this.

Reply:
Aah yep that makes sense, I'll advise adding `| undefined` as appropriate in future aws-lambda PRs but won't block them. Thanks!

Reply:
For this package in particular, events should not have it, so that `in` guards work and type machinery is simpler, and results should, so that conditionally present properties can be much more nicely defined. But as a rule of thumb, sure, you're much more likely to need it present than you are to need it not present.